packages/react/src/orders/hooks/useOrderShippingAddressChangeRequests.ts

```diff
@@ -0,0 +1,264 @@
+  | FetchOrderShippingAddressChangeRequestsFailureAction
+  | FetchOrderShippingAddressChangeRequestsRequestAction
+  | FetchOrderShippingAddressChangeRequestsSuccessAction
+  | ResetOrderShippingAddressChangeRequestsStateAction;
+
+const initialState: State = null;
+
+function reducer(state: State, action: Action) {
+  switch (action.type) {
+    case actionTypes.FetchOrderShippingAddressChangeRequestsRequest: {
+      const isSyncedWithHookParams = action.meta.isSyncedWithHookParams;
+
+      if (!isSyncedWithHookParams) {
+        return state;
+      }
+
+      const actionOrderId = action.meta.orderId;
+
+      const newState = {
+        isLoading: true,
+        error: null,
+        currentRequestId: action.meta.requestId,
+        currentOrderId: action.meta.orderId,
+        // If the action is requesting data for the same order
+        // keep the previous result on the state, otherwise
+        // set it to undefined.
+        result:
+          actionOrderId === state?.currentOrderId ? state.result : undefined,
+      };
+
+      return newState;
+    }
+    case actionTypes.FetchOrderShippingAddressChangeRequestsSuccess: {
+      // Do not update the state if:
+      // - the orderId from the action is not equal to the orderId parameter from the hook.
+      // - the orderId from the action is not equal to the orderId from the state (set on the request action dispatch).
+      // - the orderIds are the same but the requestId from the action is not the same from the state (set on the request action dispatch).
+      if (
+        !action.meta.isSyncedWithHookParams ||
+        !state ||
+        state.currentRequestId !== action.meta.requestId ||
+        state.currentOrderId !== action.meta.orderId
+      ) {
+        return state;
+      }
+
+      const newState = {
+        ...state,
+        isLoading: false,
+        result: action.payload,
+      };
+
+      return newState;
+    }
+    case actionTypes.FetchOrderShippingAddressChangeRequestsFailure: {
+      // Do not update the state if:
+      // - the orderId from the action is not equal to the orderId parameter from the hook.
+      // - the orderId from the action is not equal to the orderId from the state (set on the request action dispatch).
+      // - the orderIds are the same but the requestId from the action is not the same from the state (set on the request action dispatch).
+      if (
+        !action.meta.isSyncedWithHookParams ||
+        !state ||
+        state.currentRequestId !== action.meta.requestId ||
+        state.currentOrderId !== action.meta.orderId
+      ) {
+        return state;
+      }
+
+      const newState = {
+        ...state,
+        isLoading: false,
+        error: action.payload,
+      };
+
+      return newState;
+    }
+    case actionTypes.ResetOrderShippingAddressChangeRequestsState: {
+      return initialState;
+    }
+    default:
+      return state;
+  }
+}
+
+function useOrderShippingAddressChangeRequests(
```

Comment:
Does it make sense to add this hook to the `useOrder` hook and expose its actions from there as well?

Reply:
You are right, it does make sense. Exposed the actions on `useOrder` hook.

Reply:
Can you change the useOrder test to check if the `useOrderShippingAddressChangeRequest` hook is called with the correct parameters, please?

Reply:
Sorry, missed it. Fixed!